src/rgw/rgw_keystone.cc

```diff
@@ -149,10 +149,10 @@ KeystoneApiVersion KeystoneService::get_api_version()
   }
 }
 
-bool KeystoneToken::has_role(const string& r)
+bool KeystoneToken::has_role(const string& r) const
```

Comment:
Again, it's not new code.  Naively, it looks like this interface imposes extra (small) string copies?